vendor/github.com/inngest/inngest/pkg/util/aigateway/aigateway.go

```diff
@@ -0,0 +1,75 @@
+package aigateway
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+)
+
+// ParsedRequest represents the parsed request data for a given inference request.
+//
+// Note that this is not stored, and instead is computed just in time for each input
+// depending on the UI.
+type ParsedInferenceRequest struct {
+	URL                 url.URL  `json:"url"`
+	Model               string   `json:"model"`
+	Seed                *int     `json:"seed,omitempty"`
+	Temprature          float32  `json:"temperature,omitempty"`
+	TopP                float32  `json:"top_p,omitempty"`
+	MaxTokens           int      `json:"max_tokens,omitempty"`
+	MaxCompletionTokens int      `json:"max_completion_tokens,omitempty"`
+	StopSequences       []string `json:"stop,omitempty"`
+	// TODO: Input messages and so on.
+	// TODO: Tools
+}
+
+// ParsedInferenceResponse represents the parsed output for a given inference request.
+type ParsedInferenceResponse struct {
+	ID         string `json:"id"`
+	TokensIn   int32  `json:"tokens_in"`
+	TokensOut  int32  `json:"tokens_out"`
+	StopReason string `json:"stop_reason"`
+	// TODO: Tool use selections, parsed.
+}
+
+func ParseUnknownInput(ctx context.Context, req json.RawMessage) (ParsedInferenceRequest, error) {
+	return ParsedInferenceRequest{}, fmt.Errorf("todo")
+}
+
+// Parse validates an inference request.  This checks which model, format, and URLs we should
+// use for the given request which is passed through to the end provider.
+//
+// By default, we support the following:
+//
+// * OpenAI (and OpenAI compatible URLs, by changing the base URL)
+// * Anthropic
+// * Bedrock
+// * Google Generative AI
+// * Mistral
+// * Cohere
+// * Groq (must include URL)
+func ParseInput(ctx context.Context, req Request) (ParsedInferenceRequest, error) {
+	switch req.Opts.Format {
+	case FormatOpenAIChat:
```

Comment:
Are we going to launch with just open ai format?

Reply:
I'd _really_ like to get Anthropic in there, too, but for now I think just this.